File: data-prep/dataprep-model/src/main/scala/de/hpi/isg/dataprep/model/targets/Provenance.scala
package de.hpi.isg.dataprep.model.targets

import de.hpi.isg.dataprep.util.{Describable, Nameable}

/**
  * @author Lan Jiang
  * @since 2018/4/27
  */
abstract class Provenance
  extends Target
    with Nameable
    with Describable {

  var preparator: String
  var time: String
  var externalResourse: String
  var inputData: String
  var outputData: String
  var description: String
}

File: data-prep/dataprep-model/src/main/java/de/hpi/isg/dataprep/model/objects/Provenance.java
package de.hpi.isg.dataprep.model.objects;

import de.hpi.isg.dataprep.model.objects.system.AbstractPreparator;
import de.hpi.isg.dataprep.model.objects.system.AbstractPreparation;

/**
 * @author Lan Jiang
 * @since 2018/6/4
 */
public class Provenance extends Target {

    private AbstractPreparation preparation;
    private AbstractPreparator preparator;
    private String createdTime;
    private String messages;
}
